src/NHibernate.Test/Async/Linq/QueryLock.cs

```diff
@@ -29,7 +29,7 @@ public class QueryLockAsync : LinqTestCase
 	{
 		protected override bool AppliesTo(Dialect.Dialect dialect)
 		{
-			return TestDialect.SupportsSelectForUpdate;
+			return TestDialect.SupportsSelectForUpdate && !(dialect is SapSQLAnywhere17Dialect);
```

Comment:
Why not overriding the `TestDialect` property for it?

Reply:
Ok, I have now read the additional commit message. So, that is yet another property which would be required. But still this is strange: if it is waiting, it should end up with a timeout, shouldn't it?

Reply:
This behaviour is specific to this particular test case, so I did not want to add a property (initially I was going to add it, but decided against). The dialect itself claims that it supports "SELECT FOR UPDATE", however it does not support "nowait", which makes the check on this tests useless.

Reply:
I have debugged it. And locking in `SybaseSQLAnywhere10Dialect` is actually bugged. I think we should flag `SupportsSelectForUpdate` as false for SAP SQL anywhere.

The trouble is, it uses the `for update by lock` syntax. This syntax is cursor specific, and silently results in "no-op" (no intent/update/exclusive locks acquired) as soon as it is used in a context where the engine considers there are no updatable cursor involved.

And that is what happen: when I breakpoint after the `for update` query, there are only shared locks in the db (seen with `SELECT * FROM sa_locks();`).
To actually get `intent` lock, the query has to be changed for using `with (updlock)` hint instead of the `for update`. The hint is not bound to cursor usage and do acquire locks whatever the query or cursor used. When using such a query, then I can see locks in the database.

So the database can support locking rows, but its dialects currently fail to do so.

Reply:
Ok

Reply:
>So the database can support locking rows, but its dialects currently fail to do so.

Needs to be fixed sometime. Can you create a ticket? (I'm going to sleep now).

Reply:
#1915, good night.